ricecooker/classes/nodes.py

```diff
@@ -615,6 +619,68 @@ def validate(self):
             assert len(tag) <= 30, (
                 "ERROR: tag " + tag + " is too long. Tags should be 30 chars or less."
             )
+
```

Comment:
This was just moved up from a subclass to ensure it gets run for all node types.

Reply:
(same with the "role" stuff above)